### PR TITLE
Fixed a bug when the disk capacity was insufficient in RowFlush

### PR DESCRIPTION
**Pull request by ggtakec**
### Relevant Issue (if applicable)
n/a

### Details
I found a bug in the FdEntity::RowFlush method and fixed it.
It was forgot to start multipart upload in the process when it detected that the disk space was insufficient.
The current test was unable to detect this bug because it had to detect out of disk space just when flushing.
The test case for this will be difficult, though we need to think about it in the future.

### CODE REVIEW

File: src/fdcache_entity.cpp [hunk 1]
```diff
@@ -1224,12 +1223,22 @@ int FdEntity::RowFlush(const char* tpath, bool force_sync)
...
-                  return static_cast<ssize_t>(result);
-              }
+                // no enough disk space
+                if(nomultipart){
+                    S3FS_PRN_WARN("Not enough local storage to flush: [path=%s][fd=%d]", path.c_str(), fd);
+                    return -ENOSPC;   // No space left on device
+                }
+                if(0 != (result = NoCachePreMultipartPost())){
+                    S3FS_PRN_ERR("failed to switch multipart uploading with no cache(errno=%d)", result);
+                    return static_cast<ssize_t>(result);
```
Comment (gaul) [inline src/fdcache_entity.cpp:1233]: I think these `ssize_t` casts are not necessary.  `Load`, `NoCacheLoadAndPost`, and `NoCachePreMultipartPost` all return `int`.  I am OK merging this PR but I am concerns about the free-space handling generally.

**Review by gaul — CHANGES_REQUESTED** [review]
Could you change the typo from RawFlush to RowFlush?

@CarstenGrohmann Can you also look at this since it touches some `nomultipart` code?

File: src/fdcache_entity.cpp (hunk 1 above)
Comment (ggtakec) [reply src/fdcache_entity.cpp]: It is changed not to do static_cast and changed the same part.

This change is simply a bug in processing omissions, so it's a different issue than you're concerned about.
I agree about regarding the check of free space. I also feel that the problem is that the area size that must be checked differs depending on the upload mode.

Comment (CarstenGrohmann) [reply src/fdcache_entity.cpp]: Reading your discussion and reflecting my suggested change in #1647, I'll rework the change and move the check to `NoCachePreMultipartPost()` - it's be better placed and would also enforce the minimum size for this case.

Comment (gaul) [reply src/fdcache_entity.cpp]: Agreed that this improves the correctness.  Let's think some more about free space handling in future PRs.

Comment (ggtakec) [reply src/fdcache_entity.cpp]: Thanks

**Pull request merged**